### PR TITLE
security: refactor to support new mode `marker`.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/17206

What's Changed:

Previous work in https://github.com/tikv/tikv/pull/8670 introduces security.redact_info_log to replace the raw data by ? when logging, used to desensitize the user data.

Recently, TiDB has introduce a new "marker" mode for redact_info_log referring to https://github.com/pingcap/tidb/issues/51306 since v8.1. And when security.redact_info_log = marker, it desensitizes the user data with ‹..› .

This pr refactors the previous implementation on setting `redact_info_log` configurations and make it support the new mode `marker`, which is compatible to the implementation in TiDB.

```commit-message
Introduce a new security mode `marker` for log desensitization.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Introduce a new security mode `marker` for log desensitization.
```

### CODE REVIEW

**Comment by LykxSassinator** [comment]
/hold This pr is ready for review.

But the `marker` mode should also be implemented and supported in `proto-rust` repo. After the implementation in `proto-rust` repo is merged, this pr can be merged.

File: components/log_wrappers/src/lib.rs [hunk 1]
```diff
@@ -61,40 +60,109 @@ impl<T: std::fmt::Debug> slog::Value for DebugValue<T> {
...
 
 /// Set whether we should avoid user data to slog.
-pub fn set_redact_info_log(v: bool) {
-    REDACT_INFO_LOG.store(v, Ordering::Relaxed);
-    proto_set_redact_bytes(v);
+pub fn set_redact_info_log(config: RedactOption) {
+    let level = config.convert();
+    REDACT_INFO_LOG.store(level, Ordering::Relaxed);
+    // TODO: make this configuration suitable for protobuf.
+    proto_set_redact_bytes(level == RedactLevel::On || level == RedactLevel::Marker);
```
Comment (overvenus) [inline components/log_wrappers/src/lib.rs:165]: Any plan for adding marker support for rust-protobuf?

Comment (LykxSassinator) [reply components/log_wrappers/src/lib.rs:165]: It's on-going.

> /hold This pr is ready for review.
> 
> But the `marker` mode should also be implemented and supported in `proto-rust` repo. After the implementation in `proto-rust` repo is merged, this pr can be merged.

**Review by Connor1996 — COMMENTED** [review]
LGTM in advance

File: components/log_wrappers/src/lib.rs (hunk 1 above)
Comment (LykxSassinator) [reply components/log_wrappers/src/lib.rs:165]: Pls refer to https://github.com/pingcap/rust-protobuf/pull/11. /cc @overvenus

Comment (LykxSassinator) [reply components/log_wrappers/src/lib.rs]: All things are done for it.

**Comment by LykxSassinator** [comment]
> /hold This pr is ready for review.
> 
> But the `marker` mode should also be implemented and supported in `proto-rust` repo. After the implementation in `proto-rust` repo is merged, this pr can be merged.
> 
> PR for `rust-protobuf` is [pingcap/rust-protobuf#11](https://github.com/pingcap/rust-protobuf/pull/11)

All tasks have been completed for it. It's ready for review and merge.

**Review by overvenus — APPROVED** [review]

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Pull request merged**